Feature/#1 Add general html markup for future index page

duza opened the pull request:
Could you review this code, please?

Commit:
Add base skeleton of future index page

Fixes #1

Commit:
Add header markup prototype for future index page

Fixes #1

Commit:
Add slider prototype markup for future index page

Fixes #1

Commit:
Add special offer section for future page

Also fix missprint in title.

Fixes #1

Commit:
Add price cards sections for future index page

Fixes #1

Commit:
Add shop description section for future index page

Also fix symbol of russian ruble.

Fixes #1

Commit:
Add news section for future index page

Fixes #1

Commit:
Add footer section for future index page

Fixes #1

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+        </nav>
+        <div class="header-buttons">
+          <div class="button-search">
+            <a></a>
```
Добавь класс и описание внутри ссылки

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+            </div>  
+          </div>
+          <div class="button-enter">
+            <a>Вход</a>
```
Добавь класс

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+          </div>
+          <div class="phone">
+            <a href="tel:88124502525" class="phone-link">8 812 450-25-25</a>
+          </div>
```
время и телефон должны иметь общую обертку так как позицинироваться в дальнейшем будут вместе

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+            <div class="slider-buttons">
+              <div class="button-slide-1"><a href="#"></a></div>
+              <div class="button-slide-2"><a href="#"></a></div>
+              <div class="button-slide-3"><a href="#"></a></div>
```
это должны быть label
через атрибут for у label и id у радиобатонов установим связь и слайдер заработает

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+    <title>Gllacy Shop | Глейси - магазин мороженого</title>
+  </head>
+  <body>
+    <div class="page-container">
```
между body и page-container создай 3 <input type='radio' /> их id должен соответствовать атрибутам for у лейблов. О них я написал ниже

mdziarkach replied:
этот контейнер будет юзаться для переключение фона при переключении слайдов
если тебе нужен контейнер для центрирования то тебе придётся добавить другой, но учти что карта на всю ширину экрана должна быть

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+        <section class="container-prices">
+          <div class="price-card">
+            <img src="" alt="" class="price-img">
+            <p class="price">310&#8381;/кг</p>
```
картинка и цена должны иметь общий контейнер, т.к. цена должна будет позиционироваться относительно изображения а не всей карточки

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+            <h3>Maгазин Глэйси &mdash; это онлайн- и офлайн-магазин по продаже мороженного собственного производства на развес</h3>
+            <div class="description-container">
+              <div class="description-item">
+                <i class="item-icon"></i>
```
мы можем добавить изображения через псевдоэоемент

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+        <section class="news-container">
+          <article class="news-post">
+            <h4>Новое в нашем блоге</h4>
+            <a>10 способов сервировки фруктовых щербетов к столу</a>
```
ссылка должна куда-то вести. хотя бы на #

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+            <p>Подпишитесь на нашу сладкую рассылку и будете всегда в курсе всего самого вкусного, что у нас происходит. Обещаем не спамить и не слать всякой не нужной ерунды. Честно =)</p>
+            <form action="" method="" id="news-letter">
+              <input type="email" name="letter-email" id="letter-email" placeholder="Электронная почта">
+              <button type="submit" class="submit-button">Отправить</button>
```
обрати внимание что все кнопки у тебя имеют одинаковый вид, но классы ты почему-то задаешь им разные

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+        </div>
+      </main>
+      <footer>
+      <div class="social">
```
соблюдай вложеннойть

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,179 @@
+        <a href="#" class="facebook">Фейсбук</a>
+        <a href="#" class="vkontakt">Вконтакте</a>
+      </div>
+      <nav class="footer-nav">
```
nav обычно на странице только один - используется для основной навигации. у тебя один уже есть

duza replied:
https://html.spec.whatwg.org/multipage/sections.html#the-nav-element
с одной стороны никто не запрещает использовать несколько `nav` тэгов на одной странице, с другой - тебе виднее)

Commit:
fix: Add class and inner text to link in header

Fixes #1

Commit:
fix: Change indents for html code in index page

Fixes #1

Commit:
fix: Add common parent for working time and phone

Fixes #1

Commit:
fix: Replace div to label for relation with inputs

Also add input radio type for slider.

Fixes #1

Commit:
fix: Add centering container

Fixes #1

Commit:
fix: Add image containers for image with price

Fixes #1

Commit:
fix: Remove redundant i tags for images

We will use pseudo-elements for adding of image

Fixes #1

Commit:
fix: Add href attribute and class for fix of a tag

Fixes #1

Commit:
fix: Unify different button classes to common

Fixes #1

Commit:
fix: Replace nav in footer to simple div block

Fixes #1

Commit:
Add meta tags, styles and js files

 - Add meta tag which message browser encoding of html page.

 - Add meta tag for compatibility with older versions of ie.

 -  Add meta tag for mobile browsers.

 - Add while empty style and js files.

Fixes #1

Commit:
Add a skeleton of future catalog page

Fixes #1

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,205 @@
+  <input class="button-slide" id="button-slide-2" type="radio" name="button-slide">
+  <input class="button-slide" id="button-slide-3" type="radio" name="button-slide">
+  <div class="page-container">
+    <div class="center-container">
```
для центрирования контента блок обычно называют просто container

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,205 @@
+  <div class="page-container">
+    <div class="center-container">
+      <header>
+        <nav>
```
для навигации класс лучше задать

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,205 @@
+            <a href="#" class="search-icon">Поиск</a>
+            <div class="popup">
+              <input type="search" name="q" id="site-search" placeholder="Что ищем?">
+            </div>
```
замени div на form

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,205 @@
+              <div class="description-item">
+                <p>Все наше мороженое изготавливается на собственном производстве, с использованием современного
+                  оборудования и проверенных временем технологий.</p>
+              </div>
```
если задать класс для параграфа, а не для div то div можно будет удалить
сейчас есть лишняя обертка

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,205 @@
+      <div class="map-container"></div>
+    </section>
+    <div class="center-container">
+      <footer>
```
container лучше положить внутри footer

mdziarkach commented on index.html:
```diff
@@ -0,0 +1,205 @@
+          </ul>
+        </div>
+        <div class="copyright">
+          <img src="" alt="html academy" class="html-logo">
```
картинка скорей всего тоже является ссылкой

Commit:
fix: Rename class for centering container

Fixes #1

Commit:
fix: Add class for navigation

Fixes #1

Commit:
fix: Replace div which contains input to form

Fixes #1

Commit:
fix: Remove redundant div and add its class to p

Fixes #1

Commit:
fix: Put container inside the footer

Fixes #1

Commit:
fix: Wrap logo in a tag

Fixes #1

Commit:
fix: Move cart block from button enter block

Fixes #1

Commit:
Add header for future catalog page

Fixes #1

Commit:
Add breadcrumbs section in catalog page

- Add breadcrumbs section in catalog page

- Add active modifiers to certain blocks in header section

- Add headline

Fixes #1

Commit:
Add ice cream filters section

Fixes #1

Commit:
Add ice cream prices card grid section

Fixes #1

Commit:
Add pagination section

Fixes #1

Commit:
Add footer section for future catalog page

Fixes #1

Commit:
fix: Add submit button to enter popup

Fixes #1

Commit:
fix: Add dropdown menu in nav bar

Fixes #1

Commit:
Add prototype button cart popup with ordered good

Fixes #1

Commit:
Merge branch 'feature/#1-catalog' into feature/#1

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+<body>
+  <div class="center-container">
+    <header>
+      <nav>
```
на этой странице класс не  добавил для nav а для index.html добавил
надо к одному виду привести

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+          </li>
+          <li>
+            <a href="#">Каталог</a>
+            <div class="popup">
```
тут должен быть выпадающий ul список

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+          <li>
+            <a href="#">Мелорин</a>
+          </li>
+  </div>
```
проверь пложенность

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+    <a href="#">О Компании</a>
+  </li>
+  </ul>
+  </nav>
```
блоки вложены друг в друга а в разметке вложенность никак не видна http://prntscr.com/lowj2h

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+    <main>
+      <section class="breadcrumbs-container">
+        <div class="breadcrumbs-items">
+          <ol>
```
ol - нумерованный список. нам не нужен нумерованный список

duza replied:
Хз. Мне казалось что имеет значение порядок поэтому и ordered list https://html.spec.whatwg.org/multipage/grouping-content.html#the-ol-element Поправлю

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+    </header>
+    <main>
+      <section class="breadcrumbs-container">
+        <div class="breadcrumbs-items">
```
может class="breadcrumbs-items" перенести в ul и удалить лишнюю обертку?

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+            </div>
+            <div class="filter-item">
+              <h4 class="filter-headline">Жирность</h4>
+              <label class="radio-option" for="fat-content-1">
```
в данном случае лучше не оборачивать input в label
это помешает нам чекбоксы и радиобатоны кастомизировать нам в дальнейшем
лучше так:
```
<input />
<label></label>
```

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+        <a href="#">4</a>
+        <a href="#">5</a>
+        <a href="#">&gt;</a>
+      </section>
```
думаю лучше сделать это списком ul
```
<ul class="pagination">
  <li><a>1</a></li>
  ....
</ul>
```

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+        <a href="#" class="vkontakt">Вконтакте</a>
+      </div>
+      <div class="footer-nav">
+        <ul>
```
`<ul class="footer-nav">`
и лишнюю обертку можно удалить

mdziarkach commented on catalog.html:
```diff
@@ -0,0 +1,314 @@
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <title>Gllacy Shop | Глейси - магазин мороженого<</title> <meta name="viewport" content="width=device-width, initial-scale=1">
+      <link rel="stylesheet" type="text/css" media="screen" href="styles/main.css" />
+      <script src="js/main.js"></script>
```
поправь табуляции

mdziarkach commented on index.html:
```diff
@@ -19,180 +24,201 @@
+      <a href="#">О Компании</a>
+    </li>
+    </ul>
+    </nav>
```
вложенность поправить нужно

mdziarkach commented on index.html:
```diff
@@ -19,180 +24,201 @@
+        <div class="slider-container">
+          <div class="slide" id="slide-1">
+            <p class="slide-text">Крем-брюле и пломбир c малиновым джемом</p>
+            <a href="#" class="btn btn_big">Давайте две!</a>
```
проверь пожалуйста все кнопки. у них должен быть одинаковый класс + дополнительный класс модификатор для изменения размера

mdziarkach commented on index.html:
```diff
@@ -19,180 +24,201 @@
-            <a href="#" class="show-button">Быстрый просмотр</a>
+          <div class="slide" id="slide-3">
+            <p class="slide-text">Крем-брюле и пломбир c малиновым джемом</p>
+            <a href="#" class="btn btn_big">Давайте две!</a>
```
кнопка по идее общая и ее не обязательно в каждый слайд отдельно ложить

Commit:
fix: Add nav class for navigation in catalog page

Fixes #1

Commit:
fix: Replace div to ul tag for dropdown list

Fixes #1

Commit:
fix: Replace ol breadcrumbs to ul in catalog page

Also remove redundant container.

Fixes #1

Commit:
fix: Move radio input from labels for future

Fixes #1

Commit:
fix: Replace section tag to ul for semantics

Fixes #1

Commit:
fix: Remove redundant wrapper for footer nav

Fixes #1

Commit:
fix: Remove greater than redundant character

Also fix indents in head.

Fixes #1

Commit:
fix: Replace div to ul and fix nesting tags

Fixes #1

Commit:
fix: Remove redundant buttons and move slide btn

Fixes #1

Commit:
fix: Bring the buttons to general appearance

Fixes #1
